lib/info.dart

```diff
@@ -46,8 +50,26 @@ abstract class Info {
 // TODO(sigmund): add more:
 //  - inputSize: bytes used in the Dart source program
 abstract class BasicInfo implements Info {
+  static Set<int> _ids = new Set<int>();
   final InfoKind kind;
-  final int id;
+
+  int _id;
+  int get id {
+    if (_id == null) {
+      // we hope we have all of the parents set up by now...
+
+      var path = _getParentPath(this);
```

Comment:
Normally I'd agree, but in this case I'm thinking this helper should be merged with `longName` - we can switch `longName` to use URI for library, or we can switch it to have an extra optional argument to select whether to use the name or the uri.

`longName` is used in many command-line tools under `bin/`, and I could see us switching several of those to use the `longName` with Uris for libraries as well